seatunnel-connectors-v2/connector-paimon/src/main/java/org/apache/seatunnel/connectors/seatunnel/paimon/sink/PaimonSinkWriter.java

```diff
@@ -174,16 +175,21 @@ public void write(SeaTunnelRow element) throws IOException {
 
     @Override
     public Optional<PaimonCommitInfo> prepareCommit() throws IOException {
+        return Optional.empty();
+    }
+
+    @Override
+    public Optional<PaimonCommitInfo> prepareCommit(long checkpointId) throws IOException {
         try {
             List<CommitMessage> fileCommittables;
             if (JobContextUtil.isBatchJob(jobContext)) {
                 fileCommittables = ((BatchTableWrite) tableWrite).prepareCommit();
             } else {
                 fileCommittables =
-                        ((StreamTableWrite) tableWrite).prepareCommit(false, committables.size());
+                        ((StreamTableWrite) tableWrite).prepareCommit(false, checkpointId);
```

Comment:
Is `committables.size()` to `checkpointId` right?

Reply:
It's right. The original is always 0.